src/plugins/typechecker/simpleTestSpecification.ini

```diff
@@ -0,0 +1,85 @@
+#@META fallback/#1 = spec/examples/intKey
+fallbackIntKey = 
+
+#@META check/evenInt = _
+#@META check/int = _
+#@META check/positiveInt = _
+#@META restrict/null = _
+#@META restrict/write = _
+intKey = 5
+
+#@META check/evenInt = _
+#@META check/int = _
+#@META check/positiveInt = _
+#@META restrict/null = _
+#@META restrict/write = _
+otherIntKey = 6
+
+#@META check/int = _
+#@META fallback/#1 = spec/examples/otherIntKey
+otherFallbackIntKey = 
+
+#@META restrict/write = _
+restricted = 5
+
+#@META override/#1 = spec/examples/restricted
+overrideRestricted = 5
+
+nullary = 5
+
+#@META override/#1 = spec/examples/nullary
+#@META restrict/null = _
+nonNullarry = 6
+
+#@META check/evenInt = _
+#@META fallback/#1 = spec/examples/intKey
+otherKey = 
+
+untypedKey = 
+
+#@META fallback/#1 = spec/examples/otherKey
+#@META fallback/#2 = spec/examples/untypedKey
+transitiveFallbackKey = 
+
+#@META fallback/#1 = spec/examples/fallbackIntKey
+#@META override/#1 = spec/examples/untypedKey
+overwriteKey = 
+
+#@META check/evenInt = _
+#@META check/positiveInt = _
+#@META fallback/#1 = spec/examples/intKey
+incompatibleKey = 
+
+#@META specElektra/type = self :: a -> self :: ('TyLft 'In NonNull a)
+#@META todo = default is a haskell keyword and can't be a function name... filter for that...
+specElektra/defaultz = 
+
+#@META specElektra/type = self :: a -> self :: ('TyLft 'In NonNull a)
+specElektra/restrict/null = 
+
+#@META specElektra/type = self :: a -> self :: ('TyLft 'Out NoWrite a)
+specElektra/restrict/write = 
+
+#@META specElektra/type = self :: a -> self :: ('TyLft 'Same Encrypted a)
+#@META todo = some possibility to define 'TyLft 'Same Encrypted a as a type? type alias would be enough though
+specElektra/encrypt = 
+
+#@META specElektra/type = self :: ('TyLft 'Same Encrypted a) -> self :: a
+specElektra/decrypt = 
+
+#@META specElektra/type = self :: a -> self :: (a <+> Int)
+specElektra/check/int = 
+
+#@META specElektra/type = self :: a -> self :: (a <+> PositiveInt)
```

Comment:
maybe`elektra/spec/type`, or do we need the `elektra`. Why not `define/type`?

Reply:
Hmm. I'm not sure if define/type corresponds to what i do here. This is basically the implementation-specific detail how my type checker interprets the given checker plugin in terms of its typing rules to keep it modular, so other type checking approaches may be used as well in the future which don't necessarily need to correspond to mine. Basically, even a multi-typechecker approach would be possible in theory.

Reply:
Seems like I need more information about your plans to review this :-)

Reply:
i guess the simplest thing is to talk on our friday meeting about this, maybe i just show it to you locally afterwards, that should clear things up much faster.

Reply:
Yes, if you are not yet sure what to do it is the best to talk about it. If you know it already writing it down is better because then you practice to write your thesis 👍